tests/test_templates.py

```diff
@@ -1,26 +1,58 @@
+import os
+import re
 import shutil
 import subprocess
+import tempfile
+from collections.abc import Iterator
 from pathlib import Path
 
+import pytest
+
+commit_pattern = re.compile(r"_commit: .*")
+src_path_pattern = re.compile(r"_src_path: .*")
+tests_path = Path(__file__).parent
+root = tests_path.parent
+generated_folder = "tests_generated"
+generated_root = root / generated_folder
+
+
+@pytest.fixture(autouse=True, scope="module")
+def working_dir() -> Iterator[Path]:
+    with tempfile.TemporaryDirectory(ignore_cleanup_errors=True) as temp:
+        working_dir = Path(temp) / "template"
+        working_generated_root = working_dir / generated_folder
+        shutil.copytree(root, working_dir)
+        subprocess.run(["git", "add", "-A"], cwd=working_dir)
+        subprocess.run(
+            ["git", "commit", "-m", "draft changes", "--no-verify"], cwd=working_dir
+        )
+
+        yield working_dir
+
+        for dir_name in os.listdir(working_generated_root):
```

Comment:
https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module
can use `.iterdir()` here

Reply:
Done